src/controller/SoldierController.ts

```diff
@@ -0,0 +1,38 @@
+import { Request, Response } from 'express';
+import Soldier from '../repository/soldierRepo';
+import soldierSchema from '../repository/schemaValidation';
+
+export async function insertSoldier(req:Request, res:Response) {
+  try {
+    const validateBody = soldierSchema.parse(req.body);
+    const soldier = new Soldier(validateBody);
```

Comment:
Soldier repo is the connection to the DB, why is it a class, The class should be a part of the logic